src/libsync/discoveryphase.cpp

```diff
@@ -350,20 +349,19 @@ static std::unique_ptr<csync_file_stat_t> propertyMapToFileStat(const QMap<QStri
         } else if (property == "dDC") {
             file_stat->directDownloadCookies = value.toUtf8();
         } else if (property == "permissions") {
-            file_stat->remotePerm = value.toUtf8();
+            file_stat->remotePerm = RemotePermissions(value);
         } else if (property == "checksums") {
             file_stat->checksumHeader = findBestChecksum(value.toUtf8());
         } else if (property == "share-types" && !value.isEmpty()) {
-            // Since QMap is sorted, "share-types" is always "permissions".
-            if (file_stat->remotePerm.isEmpty()) {
+            // Since QMap is sorted, "share-types" is always after "permissions".
+            if (file_stat->remotePerm.isNull()) {
                 qWarning() << "Server returned a share type, but no permissions?";
             } else {
                 // S means shared with me.
                 // But for our purpose, we want to know if the file is shared. It does not matter
                 // if we are the owner or not.
-                // Piggy back on the persmission field 'S'
-                if (!file_stat->remotePerm.contains('S'))
-                    file_stat->remotePerm.append('S');
+                // Piggy back on the persmission field
+                file_stat->remotePerm.setPermission(RemotePermissions::IsShared);
```

Comment:
Oh, this only works because "permissions" < "share-types" (old stuff)

Reply:
yes, this is documented and enforced a few line earlier.